M0090IDE-3982 document resolve in workspace (#443)
MicroEJ/docs: SDKUserGuide/ide.rst

.. _ide:

IDE
===

.. _sdk_startup:

> SDK Startup

Reviewer: Shouldn't it be "IDE Startup", or even just "Startup" ?

Author: The section has been renamed.

-----------

When starting the SDK, it prompts you to select the last used workspace or a default workspace on the first run. 

.. figure:: images/workspace_selection.png
   :alt:  Workspace selection
   :align: center
   :scale: 100%

   Workspace selection

A workspace is a main folder where to find a set of projects containing MicroEJ source code.

When loading a new workspace, the SDK prompts for the location of the MicroEJ repository, 
where Architectures, Platforms or Virtual Devices will be imported. 

.. figure:: images/repository_selection.png
   :alt:  Repository selection
   :align: center
   :scale: 100%

   Repository selection

By default, the SDK suggests to point to the default MicroEJ repository on your operating system, 
located at ``${user.home}/.microej/repositories/[version]``. 
You can select an alternative location. Another common practice is to define a local
repository relative to the workspace, so that the workspace is
self-contained, without external file system links and can be shared
within a zip file.

.. _resolve_dependencies_in_workspace:

Resolve dependencies in workspace

> ---------------------------------

Reviewer: This section should be moved in a dedicated page.

Author: According to our discussion, all the documentation related to the IDE had to be merged into a single page "IDE", so why should this section be moved ?

Reviewer: No, all the documentation related to the IDE must be gathered under an "IDE" level, but each one in a dedicated page. Otherwise the page would be huge since we will continue to add content.

Author: The "IDE" level has been added

.. _ide:

IDE
===

> The SDK provides an Integrated Development Environment (IDE) for creating MicroEJ Applications and building

Reviewer 2: ```suggestion
The SDK provides an Integrated Development Environment (IDE) for creating and building Applications
```

.. _ide:

IDE
===

> **SDK Version 5.x**, embedded in MicroEJ SDK, is an Integrated Development Environment (IDE) for creating MicroEJ Applications and building

Reviewer: ```suggestion
The SDK provides an Integrated Development Environment (IDE) for creating MicroEJ Applications and building
```

> MicroEJ Firmware. The SDK is based on Eclipse Java Edition and relies on the integrated Java Compiler (JDT).

Reviewer: ```suggestion
MicroEJ Firmware. It is based on Eclipse Java Edition and relies on the integrated Java Compiler (JDT).
```

The SDK is also packaged with Eclipse to produce a SDK distribution. To download the latest distribution, please consult

> https://developer.microej.com/microej-sdk-software-development-kit/.

Reviewer: The sentence should be removed IMO, it is related to the whole Distribution.

Author: Yes, I didn't see that the link to download the SDK was already indicated in https://docs.microej.com/en/latest/SDKUserGuide/install.html, so I added it here. The sentence has been removed.
